src/core/undo/InsertLayerUndoAction.cpp

```diff
@@ -9,10 +9,12 @@
 #include "util/i18n.h"
 
 InsertLayerUndoAction::InsertLayerUndoAction(LayerController* layerController, const PageRef& page, Layer* layer,
-                                             int layerPosition):
-        UndoAction("InsertLayerUndoAction"), layerPosition(layerPosition), layerController(layerController) {
+                                             size_t layerPosition):
+        UndoAction("InsertLayerUndoAction"),
+        layerPosition(layerPosition),
+        layerController(layerController),
+        layer(layer) {
     this->page = page;
```

Comment:
```suggestion
        layer(layer). 
        page(page) {
```

Reply:
No can do without refactoring the undo business. `page` is a member of the base class UndoAction (and some derived classes don't set it (e.g. `EmergencySaveRestore`). Changing this would go to deep for this PR.

Reply:
Ok, I didn't know that this is a base class member.